Sources/SnapshotTesting/Common/ForwardingWKNavigationDelegate.swift

```diff
@@ -0,0 +1,133 @@
+  func webView(
+    _ webView: WKWebView,
+    decidePolicyFor navigationAction: WKNavigationAction,
+    preferences: WKWebpagePreferences,
+    decisionHandler: @escaping (WKNavigationActionPolicy, WKWebpagePreferences) -> Void
+  ) {
+    original?.webView?(
+      webView,
+      decidePolicyFor: navigationAction,
+      preferences: preferences,
+      decisionHandler: { [weak self] policy, prefs in
+        decisionHandler(policy, prefs)
+        if policy == .cancel {
+          self?.finish()
+        }
+      }
+    )
+  }
+
+  func webView(
+    _ webView: WKWebView,
+    decidePolicyFor navigationAction: WKNavigationAction,
+    decisionHandler: @escaping (WKNavigationActionPolicy) -> Void
+  ) {
+    original?.webView?(webView, decidePolicyFor: navigationAction, decisionHandler: { [weak self] policy in
+      decisionHandler(policy)
+      if policy == .cancel {
+        self?.finish()
+      }
+    })
+  }
+
+  func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+    original?.webView?(webView, didFinish: navigation)
+    webView.evaluateJavaScript("document.readyState") { [weak self] _, _ in
+      self?.finish()
+    }
+  }
+
+  func webView(_ webView: WKWebView, didFail navigation: WKNavigation!, withError error: Error) {
+    original?.webView?(webView, didFail: navigation, withError: error)
+    finish()
+  }
+
+  func webView(_ webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: Error) {
+    original?.webView?(webView, didFailProvisionalNavigation: navigation, withError: error)
+    finish()
+  }
+
+  @available(OSXApplicationExtension 10.11, *)
+  func webViewWebContentProcessDidTerminate(_ webView: WKWebView) {
+    original?.webViewWebContentProcessDidTerminate?(webView)
+    finish()
+  }
+}
+
+fileprivate let decidePolicyPre13 = #selector(WKNavigationDelegate.webView(_:decidePolicyFor:decisionHandler:) as (
+  (WKNavigationDelegate) ->
+  (WKWebView, WKNavigationAction, @escaping (WKNavigationActionPolicy) -> Void) -> Void
+)?)
+
+@available(iOSApplicationExtension 13.0, *, OSXApplicationExtension 10.15, *)
+fileprivate let decidePolicy13 = #selector(WKNavigationDelegate.webView(_:decidePolicyFor:preferences:decisionHandler:))
+
+fileprivate var interceptedSelectors: Set<Selector> = {
+  var selectors: Set<Selector> = [
+    #selector(WKNavigationDelegate.webView(_:didFinish:)),
+    #selector(WKNavigationDelegate.webView(_:didFail:withError:)),
+    #selector(WKNavigationDelegate.webView(_:didFailProvisionalNavigation:withError:)),
+    decidePolicyPre13,
+  ]
+
+  if #available(iOSApplicationExtension 13.0, *, OSXApplicationExtension 10.15, *) {
+    selectors.insert(decidePolicy13)
+  }
+
+  if #available(iOSApplicationExtension 8.0, *, OSXApplicationExtension 10.11, *) {
+    selectors.insert(#selector(WKNavigationDelegate.webViewWebContentProcessDidTerminate(_:)))
+  }
+  return selectors
+}()
+
+#endif
```

Comment:
If you want 100% coverage, we should add another test file for this class that fully tests this code without the snapshots. Please let me know, happy to provide it.